src/context.ts

```diff
@@ -285,6 +285,24 @@ const setDefaultContentType = (contentType: string, headers?: HeaderRecord): Hea
   }
 }
 
+/**
+ * This function ensures that the string is safely encoded,
+ * and not double-encoded.
+ * The input string may or may not escaped by encodeURI.
+ */
+const safeEncodeURI = (str: string) => {
```

Comment:
I was a little unsure about this, but I think it would be better to move `safeEncodeURI` to `src/utils/url.ts`. It is only used in `src/contex.ts`, so it is not necessary, but since all other functions related to URL/URI are in that file, it would be better.

In that case, writing unit tests for `safeEncodeURI` in `src/utils/url.test.ts` will be good.

Reply:
Cool, I don't have any preference of that, moved